### PR TITLE
docs: Add a technical example showing scikit-learn API compatible estimators that work with skore

### PR DESCRIPTION
**Pull request by sylvaincom**
Fix https://github.com/probabl-ai/skore/issues/1451

Supersedes https://github.com/probabl-ai/skore/pull/1814 that is too long but archived for safe-keeping

### CODE REVIEW

File: examples/technical_details/plot_sklearn_api.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+Using skore with scikit-learn compatible estimators
+=====================================================================================
+
+This example shows how to use skore with scikit-learn compatible estimators.
+For these models, skore provides a unified interface for reporting and evaluation.
+Any model that can be used with the scikit-learn API can be used with skore.
+Skore's :class:`~skore.EstimatorReport` can be used to report on any estimator
+that has a ``fit`` and ``predict`` method.
+In fact, skore only requires the ``predict`` method if the estimator has already
+been fitted.
```
Comment (MarieSacksick) [inline examples/technical_details/plot_sklearn_api.py]:
```suggestion
This example shows how to use skore with scikit-learn compatible estimators.

Any model that can be used with the scikit-learn API can be used with skore.
Skore's :class:`~skore.EstimatorReport` can be used to report on any estimator
that has a ``fit`` and ``predict`` method.
In fact, skore only requires the ``predict`` method if the estimator has already
been fitted.
```
To simplify, otherwise it's both redundant and misleading: "for these models", sounds like not for the others.

File: examples/technical_details/plot_sklearn_api.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+.. note::
+
+  When using an :class:`~skore.EstimatorReport` to compute the ROC AUC and ROC curve
+  for a classification task, the estimator must have a ``predict_proba`` method.
```
Comment (MarieSacksick) [inline examples/technical_details/plot_sklearn_api.py]:
```suggestion
  When computing the ROC AUC or ROC curve
  for a classification task, the estimator must have a ``predict_proba`` method.
```
because it will be the case for comparison and crossvalidation reports too.

File: examples/technical_details/plot_sklearn_api.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+In this example, we showcase a gradient boosting model
+(`XGBoost <https://github.com/dmlc/xgboost>`_) and a custom estimator.
+
+.. note::
```
Comment (MarieSacksick) [inline examples/technical_details/plot_sklearn_api.py]: I don't think the highlight with note is required here. It attracts a lot the attention of the user, while it's just a list.

File: examples/technical_details/plot_sklearn_api.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+from sklearn.datasets import make_classification
+
+X, y = make_classification(
+    n_samples=1_000,
+    n_features=5,
+    n_classes=2,
+    class_sep=0.3,
+    n_clusters_per_class=1,
+    random_state=42,
+)
```
Comment (MarieSacksick) [inline examples/technical_details/plot_sklearn_api.py]: As in the previous examples, let's keep it simple here: you can remove at the very least the `n_classes` argument, and I would argue everything except the random_state: we want to show the interface, at no moment the actual performance.

File: examples/technical_details/plot_sklearn_api.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+print(f"{X.shape = }")
+
+# %%
+# We split our data:
+
+# %%
+from skore import train_test_split
+
+split_data = train_test_split(X, y, shuffle=True, random_state=42, as_dict=True)
```
Comment (MarieSacksick) [inline examples/technical_details/plot_sklearn_api.py]: why is shuffle explicitely True, since it's the default, plus it triggers a warning?  Let's either remove it, or pass it to False.

File: examples/technical_details/plot_sklearn_api.py [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+#
+# For this binary classification task, we consider a gradient-boosted decision trees
+# model from a library external to scikit-learn.
+# One of the most popular is `XGBoost <https://github.com/dmlc/xgboost>`_.
+
+# %%
+from skore import EstimatorReport
+from xgboost import XGBClassifier
+
+xgb = XGBClassifier(n_estimators=50, max_depth=3, learning_rate=0.1, random_state=42)
```
Comment (MarieSacksick) [inline examples/technical_details/plot_sklearn_api.py:76]: when hovering `XGBClassifier`, the docs redirects to https://scikit-learn.org/stable/modules/generated/sklearn.base.ClassifierMixin.html#sklearn.base.ClassifierMixin; it should redirect to https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBClassifier instead, if anything.

image

File: examples/technical_details/plot_sklearn_api.py [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        check_is_fitted(self)
+        X = validate_data(self, X, reset=False)
+        closest = np.argmin(euclidean_distances(X, self.X_), axis=1)
+        return self.y_[closest]
+
+
+# %%
+# .. note::
+#
+#   The estimator above does not have a `predict_proba` method.
```
Comment (MarieSacksick) [inline examples/technical_details/plot_sklearn_api.py]:
```suggestion
# .. note::
#
#   The estimator above does not have a `predict_proba` method, therefore 
#   we cannot display the ROC as above.
```

File: examples/technical_details/plot_sklearn_api.py [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+# %%
+custom_report = EstimatorReport(CustomClassifier(), pos_label=1, **split_data)
+custom_report.metrics.precision()
+
+# %%
+# Conclusion
+# ==========
+#
+# This example demonstrates how skore can be used with scikit-learn compatible estimators.
+# Any estimator that follows the scikit-learn API can be integrated with skore for evaluation and reporting.
```
Comment (MarieSacksick) [inline examples/technical_details/plot_sklearn_api.py]:
```suggestion
# This example demonstrates how skore can be used with scikit-learn compatible estimators.
```

The second sentence is redundant.

**Review by MarieSacksick — CHANGES_REQUESTED** [review]

File: examples/technical_details/plot_sklearn_api.py (hunk 3 above)
Comment (sylvaincom) [reply examples/technical_details/plot_sklearn_api.py]: Ok, done

File: examples/technical_details/plot_sklearn_api.py (hunk 4 above)
Comment (sylvaincom) [reply examples/technical_details/plot_sklearn_api.py]: Ok, done

File: examples/technical_details/plot_sklearn_api.py (hunk 5 above)
Comment (sylvaincom) [reply examples/technical_details/plot_sklearn_api.py]: Ok, done

File: examples/technical_details/plot_sklearn_api.py (hunk 6 above)
Comment (sylvaincom) [reply examples/technical_details/plot_sklearn_api.py:76]: Ok, done

**Review by MarieSacksick — APPROVED** [review]
Thanks, good for me :)!
